crates/oxc_linter_plugin/src/plugin.rs

```diff
@@ -0,0 +1,615 @@
+                                .expect("for int64 number from trustfall to fit into i32 for js");
+                            Value::Number(as_i32.into())
+                        }
+                        FieldValue::Uint64(int) => {
+                            let as_u32: u32 = int
+                                .try_into()
+                                .expect("for Uint64 number from trustfall to fit into u32 for js");
+                            Value::Number(as_u32.into())
+                        }
+                        FieldValue::Float64(f64) => Value::Number(f64),
+                        FieldValue::String(str) => Value::String(mv8.create_string(&str)),
+                        FieldValue::List(list) => {
+                            let arr = mv8.create_array();
+                            for el in &*list {
+                                arr.push(to_v8(mv8, el.clone())).expect(
+                                    "to be able to put elements from trustfall list into js list",
+                                );
+                            }
+                            Value::Array(arr)
+                        }
+                        _ => unreachable!(),
+                    }
+                }
+                data.set(k.to_string(), to_v8(mv8, v)).unwrap();
+            }
+
+            let fn_return: Value =
+                transformer.call(Values::from_vec(vec![Value::Object(data)])).unwrap();
+
+            let object_returned = match fn_return {
+                Value::Null => continue,
+                Value::Object(object) => object,
+                _ => unimplemented!(
+                    "You should not return values that aren't objects or nulls from js."
+                ),
+            };
+
+            let transformed_data_to_span: SpanInfo = match (
+                object_returned.get("span_start").unwrap(),
+                object_returned.get("span_end").unwrap(),
+            ) {
+                (Value::Number(a), Value::Number(b)) => SpanInfo::SingleSpanInfo(SingleSpanInfo {
+                    span_start: a as u64,
+                    span_end: b as u64,
+                    fix: match object_returned.get("fix").unwrap() {
+                        Value::String(fix) => Some(fix.to_string()),
+                        _ => None,
+                    },
+                    summary: match object_returned.get("summary").unwrap() {
+                        Value::String(summary) => Some(summary.to_string()),
+                        _ => None,
+                    },
+                    reason: match object_returned.get("reason").unwrap() {
+                        Value::String(reason) => Some(reason.to_string()),
+                        _ => None,
+                    },
+                }),
+                (Value::Array(a), Value::Array(b)) => {
+                    let a_count = a.len();
+                    SpanInfo::MultipleSpanInfo(MultipleSpanInfo {
```

Comment:
This is also not readable: the `MultipleSpanInfo` struct literal is huge and has `match` statements inside.